### PR TITLE
Allow installing in folder with space in name

### PR DESCRIPTION
**Pull request by Et7f3**
Hello,

As part of split of #10727 here a first step.
https://github.com/ocaml/ocaml/pull/10727/files#r748650610 comment associated.

This allow to install man page and be able to see it:
```
./configure --prefix="$PWD/install here"
make
make install
export MANPATH="$PWD/install here"/man
man Lazy # We get the recent version
```
check-stale allow finer message. So it it a improvement on itself and it works with space in path.

### CODE REVIEW

**Comment by shindere** [comment]
This seems correct to me, but I'd like @dra27's review, please.

File: Changes [hunk 1]
```diff
@@ -130,6 +130,8 @@ Working version
   (Gabriel Scherer, review by Luc Maranget and Vincent Laviron)
 
 ### Build system:
+- #11590: Allow installing in folder with space in nameh
```
Comment (dra27) [inline Changes]: There's a typo on "name", but possibly instead:
```suggestion
- #11590: Allow installing to a destination path containing spaces.
```
?

**Review by dra27 — COMMENTED** [review]
The fix to `man/Makefile` is clearly correct (thanks!)

Other changes we've been making have been trying to move away from using shell constructs and the dancing with bash-isms, not to mention having to worry about locales at all, makes me very nervous about the other change. We have been stung - at considerable loss of time both to the original reporter and to those who had to reproduce and fix it - by the vagaries of shell commands being sensitive to user locales, and GNU make (like ISO C!) has the benefit of being evaluated in the C locale.

The test can fairly easily be kept in `make`, just by taking advantage of `make -C <dir> -f <Makefile>` to do the change of directory. I tested the idea in https://github.com/dra27/ocaml/commit/83b12383ec0222bc098315791a217d2f1eb6dd6b which works both on Windows and macOS (I installing this over the top of a 4.12 installation). The crux of it is that the `install::` target becomes:

```make
.PHONY: install
# Ensure any pre-4.13 lowercased artefacts are removed on macOS and Windows
install::
	@$(MAKE) -C '$(INSTALL_LIBDIR)' -f '$(realpath Makefile.check-stale)' \
	         EXT=.cmi \
	         'ARTEFACTS=$(wildcard stdlib__*.cmi)' \
	         'INSTALL_LIBDIR=$(INSTALL_LIBDIR)'
```

which makes a recursive call _in the target directory_ and the GNU make definition for `$(stale)` is expanded into `MAkefile.check-stale` as:

```make
INSTALLED_ARTEFACTS = $(notdir $(wildcard stdlib__*$(EXT)))
STALE_ARTEFACTS = $(filter-out $(ARTEFACTS), $(INSTALLED_ARTEFACTS))
endif
```

and now `$(STALE_ARTEFACTS)` will be non-empty if any files with the old casing are found. The "trick" is that using `-C` avoids the need for specifying paths with spaces in them inside a `Makefile` (which is impossible)

**Comment by Et7f3** [comment]
Ok so you want I commit your code with you as co-author ?

**Comment by dra27** [comment]
If you're happy with that commit, sure - the main thing is that I think the check should be done using GNU make rather than using the shell, I just didn't want to suggest that before being sure that the solution actually worked (which is why I wrote the commit!)

**Comment by Et7f3** [comment]
EXT=.cmi instead of .cm* is it for a reason ? Inlining it in the wildcard should be fine ?

**Comment by dra27** [comment]
> EXT=.cmi instead of .cm* is it for a reason ? Inlining it in the wildcard should be fine ?

As in https://github.com/ocaml/ocaml/pull/10301/files#r746338143 🙂

**Comment by Et7f3** [comment]
A yes. Thx for reminder (it has been more than one year).

**Comment by Et7f3** [comment]
Since `installopt-default` and `install` depend on this staleness check I keep the auxiliary target. I removed echo bashim with makefile error command and if to check if the variable is empty.

A `HACKTOBERFEST-ACCEPTED` label is appreciated if you consider this PR non-spammy.

**Comment by dra27** [comment]
Sorry for the delay getting back to you, @Et7f3 - there's consensus that we can take an even simpler route with `stdlib/Makefile` and just delete the check for mis-cased artefacts completely! Would you be happy to make that change part of this PR? (it just means deleting entirely LL57-67, re-adding the `.PHONY: install` above L70 and deleting LL97-106)

**Comment by Et7f3** [comment]
CI is green

File: Changes [hunk 2]
```diff
@@ -130,7 +130,7 @@ Working version
   (Gabriel Scherer, review by Luc Maranget and Vincent Laviron)
 
 ### Build system:
-- #11590: Allow installing to a destination path containing spaces.
+- #11590: Delete the check for mis-cased artefacts
```
Comment (dra27) [inline Changes]: I don't think change is necessary - the PR still primarily allows installation to a destination path containing spaces!

Comment (Et7f3) [reply Changes]: Ok old commit 6ed82b5f06196ef1922eaab05a5787acdad99ab9 I just reverted Changes

**Review by dra27 — APPROVED** [review]
Good to go with CI, thanks!

Thanks for your patience with this one - I’m reasonably confident that the next ones won’t take quite so much “to and fro” 🙂

**Comment by kit-ty-kate** [comment]
Is there any chance this could be backported to the 4.14 and 5.0 branches?

**Comment by dra27** [comment]
Certainly fine for the `man/Makefile` change - not instantly sure about the `stdlib/Makefile` change (it removes the test rather than fixing it)

**Comment by Et7f3** [comment]
We have found a version that test and doesn't fail with space in path. Do you want I restore it ?

**Comment by dra27** [comment]
No, it's OK, thanks - we certainly won't back-port a fix we didn't merge! @Octachron - can I manually push the `man/Makefile` change to 4.14+5.0 (possibly further, for cases where that's the only blocking issue for installing to directories with spaces)

**Comment by Octachron** [comment]
The `man` changes seems very fine to backport directly to 4.14 and 5.0 to me.

**Comment by dra27** [comment]
Pushed to both, thanks!

**Pull request merged**